pkg/k8s/watchers/pod.go

```diff
@@ -731,6 +742,27 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	return nil
 }
 
+func deepEqualBackends(backends1 []*loadbalancer.Backend, backends2 []*loadbalancer.Backend) bool {
+	if len(backends1) != len(backends2) {
+		return false
+	}
+
+	l3n4AddrMap := make(map[string]*loadbalancer.Backend)
+
+	for _, backend1 := range backends1 {
+		l3n4AddrMap[backend1.L3n4Addr.AddrCluster.String()] = backend1
+	}
+
+	for _, backend2 := range backends2 {
+		if backend1, ok := l3n4AddrMap[backend2.L3n4Addr.AddrCluster.String()]; ok && backend1.L3n4Addr.DeepEqual(&backend2.L3n4Addr) {
```

Comment:
Why call the `DeepEqual` method if there was a look up done based on the `L3nL4Addr` string?

Reply:
Good point. Could the `L4Addr`  or `Scope` in the `L3n4Addr` differ for two backend with the same IP? If not, i think we could definitely get rid of the `DeepEqual` check 👍

Reply:
Done!